qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/compact/bitmap/Bitmap375Big.java

```diff
@@ -460,11 +461,67 @@ public static long binarySearch(LongArray arr, long val) {
 		return min;
 	}
 
+	public static long binarySearchNew(LongArray arr, long val) {
+		long min = 0;
+		long max = arr.length();
+		long mid;
+
+		long[] prevFound = arr.getPrevFound();
+
+		int index = (int) (val / 65536 + 1);
```

Comment:
We can store the bucket size (65536) in the LongArray and make it implementation specific.